### ISSUE TITLE
ui-sref-opts to accept notify option

### ISSUE THREAD

**Issue by intellix** [angular-ui/ui-router#1875]
I've got tabs with a fairly large dataset and when you change tab, the old is transitioned back/forward. Anyway, it's quite heavy for me to notify the controller and re-process things. So I removed these categories from the URL and just use `ng-click` to change category. After using pre-render I see that the other categories aren't linked to/accessible so I'm going to need URLs. I've stumbled upon the `notify` option on $state.go and this allows the URL to change when I click a link... but there's still no href on my links for prerender to pickup because it happens within an ng-click.

Basically, I want to achieve the following:
```html
<a ui-sref="things({ slug: category.slug })" ui-sref-opts="{notify: false}" ng-click="setCategory(category.slug)">{{category.name}}</a>
``` 

```js
angular.module('app').controller('ThingsCtrl', function($scope, $stateParams, ThingService) {

    ThingService.getCategories().then(function(categories) {
        if ($stateParams.slug) {
            ThingService.setCategory($stateParams.slug);
        }
    });

    $scope.setCategory = function(category)
    {
        ThingService.setCategory(slug);
    };
});
```
This way I get my performance, transitions and prerender picks up hrefs so you can go directly to a category.

Is there any reason why `notify` isn't whitelisted as an option? On that note, is there a reason why we need to restrict usage of $state.go? As it stands, I have to fork and whitelist it.

**Comment by nateabele** [comment]
> Is there any reason why notify isn't whitelisted as an option? On that note, is there a reason why we need to restrict usage of $state.go? As it stands, I have to fork and whitelist it.

If I recall correctly, things worked a little differently when it was originally introduced, and certain options could have had the potential to cause weird behavior. Now I don't know, we may not need it. Feel free to submit a PR, and be sure to follow the contributor guidelines.

**Comment by tamtakoe** [comment]
:+1: Merge PR, please

**Comment by eddiemonge** [comment]
fixed in #2187
